### ISSUE TITLE
Making conda rich-er

### ISSUE THREAD

**Issue by ForgottenProgramme** [conda/conda#12766]
### Summary

This epic holds all work and discussions around adding [Rich](https://github.com/Textualize/rich) to conda. 

### User impact

When this epic has been completed, users will have a new CLI interface they can use based on rich components. We hope this allows for an improved experience. The current CLI interface will not go away and will still be available.

### Spike issues

These issues will be dedicated to researching how rich might look and feel for an application like conda. The work will mostly consist of wire frame designs so we can reach consensus before undertaking the actual work.

- [x] #13214
- [x] #13215
- [x] #13216
- [x] #13217
- [x] #13226

### conda-rich mockups

This repository contains some mockups for various conda commands: [ForgottenProgramme/rich-conda-mocks](https://github.com/ForgottenProgramme/rich-conda-mocks). These mock ups give a reasonable starting point for completing the prototypes listed above.

### Linked Issues & PRs

- [x] https://github.com/conda/conda/issues/11495
- [ ] https://github.com/conda/conda/issues/12210

### Linked epics

- [ ] https://github.com/conda/conda/issues/13707

**Comment by dholth** [comment]
Generate a threadpool of workers that simulate many parallel downloads. Have the threadpool emit progress events to a progress bar system. Ensure that the progress bars can update, complete, and make room for additional progress bars as download workers complete.

**Comment by travishathaway** [comment]
@dholth,

Please feel free to create a spike issue specifically for progress bars if that is something you would like to have prototyped. I think it would be nice to not only have package download progress bars but repodata progress bars too.

**Comment by jezdez** [comment]
I've filed #13226 to make sure we're thinking about the best practices and needed code infra to make such a switch possible and also future-proof (e.g. in case rich is stopped being maintained).

**Comment by ForgottenProgramme** [comment]
I created a repo to store mocks that I created using rich.
Here: https://github.com/ForgottenProgramme/rich-conda-mocks

**Comment by ForgottenProgramme** [comment]
This document stores insights from the research that I did to see how other big open source projects are using rich:
https://docs.google.com/document/d/1am8gyz5OYjMX81Lk3ZPJ_7GSOtrBTWoNlPUSFU2nZS4/edit?usp=sharing

**Comment by jezdez** [comment]
> This document stores insights from the research that I did to see how other big open source projects are using rich: [docs.google.com/document/d/1am8gyz5OYjMX81Lk3ZPJ_7GSOtrBTWoNlPUSFU2nZS4/edit?usp=sharing](https://docs.google.com/document/d/1am8gyz5OYjMX81Lk3ZPJ_7GSOtrBTWoNlPUSFU2nZS4/edit?usp=sharing)

Please transfer the content into a public issue or leave as a comment, for long-term storage

**Comment by ForgottenProgramme** [comment]
**Rich Best Practices Research**

TLDR:
Twine and Pytorch Lightning rely entirely on `rich` to display data on CLI. 
Streamlit only uses rich to log exception tracebacks. This too is optional and is set to FALSE by default. Apache-airflow has written custom classes using `Console` and `Table` classes from `rich`. Scipy is using `rich-click` and airflow uses `rich-argparse`.


**Popular Projects that use `rich`:**

1. [Streamlit](https://github.com/streamlit/streamlit)

- Streamlit uses' rich' to log exception tracebacks with highlighting and syntax formatting. 
- It is optional and defaults to False (in which case Python’s default tracebacks are used)
- See [code](https://github.com/streamlit/streamlit/blob/899cbbdabb1d57c1e35d195a63a180557bb54876/lib/streamlit/config.py#L442)

2. [Pytorch Lightning](https://github.com/Lightning-AI/pytorch-lightning)

- Pytorch Lightning is using `rich` for their CLI as a default (it’s hardcoded) and not something users can opt into
- See [code](https://github.com/search?q=repo%3ALightning-AI%2Fpytorch-lightning%20%22from%20rich%22&type=code)


3. [PyPA/twine](https://github.com/pypa/twine)

- Twine also uses `rich` hard coded. 
- They started using rich for progress bars instead of tqdm starting in 2022


4. [apache/airflow](https://github.com/apache/airflow)

- [`class AirflowConsole(Console)`](https://github.com/apache/airflow/blob/81be6ac6c091c9c922c4060b12fc92e2d7ad424d/airflow/cli/simple_table.py#L41C2-L41C2)
- They have a custom class derived from the Console class of rich. 
- Similarly, they have `class SimpleTable(Table)` derived from Table class of rich.


5. [ray-project/ray](https://github.com/ray-project/ray) 

- Their [RLlib CLI](https://github.com/ray-project/ray/blob/8af9a4086b3c3fce4b1fe0d1c05d2a4add0586e5/doc/source/rllib/rllib-cli.md?plain=1#L350) uses rich formatting.

**Additional:**

1. [Scipy](https://github.com/scipy/scipy/blob/8e9d43838b7db8d40fc4567f3aafeed68466b464/dev.py#L151) is using “[rich-click](https://pypi.org/project/rich-click/0.3.0/)”
2. [apache/airflow](https://github.com/apache/airflow) is using “[rich-argparse](https://pypi.org/project/rich-argparse/)” also
`class AirflowHelpFormatter(RichHelpFormatter)` They are using the RichHelpFormatter provided by rich_argparse and creating a custom class using it.

**Comment by travishathaway** [comment]
We now have a [conda-rich](https://github.com/conda-incubator/conda-rich) project under the [conda-incubator](https://github.com/conda-incubator) organization on GitHub.

I'm going to close this epic so we can move all discussion for feature implementation to that repository's issue tracker.
